unit_testing/test_reports.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# pytest unit tests for spinalcordtoolbox.reports
+
+import pytest
+import numpy as np
+
+from spinalcordtoolbox.image import Image
+from spinalcordtoolbox.reports.slice import Sagittal
+
+
+def prepare_labeled_data():
+    """Generate image/label pairs for various test cases of
+    test_sagittal_slice_get_center_spit."""
+    # Base labeled image
+    im_seg = Image('sct_testing_data/t2/labels.nii.gz')
+
+    # Create image with all but one label removed
+    im_seg_one_label = im_seg.copy()
+    for x, y, z in np.argwhere(im_seg_one_label.data)[1:]:
+        im_seg_one_label.data[x, y, z] = 0
+
+    # Create image with no labels
+    im_seg_no_labels = im_seg.copy()
+    for x, y, z in np.argwhere(im_seg_no_labels.data):
+        im_seg_no_labels.data[x, y, z] = 0
+
+    return [im_seg, im_seg_one_label, im_seg_no_labels]
```

Comment:
I wanted to use this image list with [`pytest.mark.parametrize`](https://docs.pytest.org/en/stable/parametrize.html). 

But, this usage seems to be prevented by this long-standing highly-requested issue, https://github.com/pytest-dev/pytest/issues/349.

Reply:
https://github.com/pytest-dev/pytest/issues/349#issuecomment-112203543 suggests calling it as a function like this:

```python
@pytest.mark.parametrize('im_seg', generate_labeled_data())
def test_sagittal_slice_get_center_spit(im_seg):
```

But, this creates a weird interaction with `xdist`/`download_data`, when more than 1 worker is used. Sometimes it works, but sometimes redownloading the data causes a FileNotFound error, because the data folder will be wiped for each worker. (See attached [log.txt](https://github.com/neuropoly/spinalcordtoolbox/files/5743237/xdist_download-data_filenotfound-error_log.txt).) This is related to #2957 and #2959.

I can fix this by setting `-n 1` as recommended in https://github.com/neuropoly/spinalcordtoolbox/issues/2957#issuecomment-729699201, but I'm not sure what the behavior will be like in the CI.

Reply:
I've implemented this in https://github.com/neuropoly/spinalcordtoolbox/pull/3116/commits/78382101598696cb9033ba2eefb2180eec186942. It's probably a bit overkill for such a simple bug in a minor function, though. :sweat_smile: